### ISSUE TITLE
Add simple ResourceVersion tracking in fake.Clientset

### ISSUE THREAD

**Issue by stippi2** [kubernetes/client-go#1385]
Currently, the `fake.Clientset` does not maintain any ResourceVersions on the objects that are added via it's API functions. For example:

```go
client := fake.NewClientset()
namespaces := clientset.CoreV1().Namespaces()

// Create a namespace via the API:
ns := &corev1.Namespace{
	ObjectMeta: metav1.ObjectMeta{
		Name: "test-namespace",
	},
}
_, _ := namespaces.Create(ctx, ns, metav1.CreateOptions{})

// Retrieve the namespace by name
ns, _ = namespaces.Get(context.TODO(), "test-namespace", metav1.GetOptions{})

// ns.ResourceVersion == ""
```

In the above example, the `ResourceVersion` of the object is not maintained by the fake client. For some type of tests, it would be a great benefit to update the resource version in the fake client.

We open this issue to get feedback whether this feature is considered valuable. If so, we offer to implement it. We saw there was already an [earlier issue](https://github.com/kubernetes/client-go/issues/539) that got closed due to lack of activity.

We have successfully implemented the resource version updates by registering a Reactor that watches the Create, Update and Patch actions. But it feels like it would not hurt to have this functionality directly in client-go. What do you think?

**Comment by mostafaCamel** [comment]
Hi, @stippi @stippi2 could you share your implementation of watching the PatchAction? I am having trouble implementing it for my project as `PatchActionImpl` does not have a `GetObject()` method (while CreateActionImpl and UpdateActionImpl have this method)

https://github.com/kubernetes/client-go/blob/387edb880f47ba8dcb85da09aa916d0c10fee37a/testing/actions.go#L728

**Comment by stippi2** [comment]
> Hi, [@stippi](https://github.com/stippi) [@stippi2](https://github.com/stippi2) could you share your implementation of watching the PatchAction? I am having trouble implementing it for my project as `PatchActionImpl` does not have a `GetObject()` method (while CreateActionImpl and UpdateActionImpl have this method)

Yes, [this](https://github.com/stippi2/kubernetes-replicator/blob/use-fake-client/replicate/secret/secrets_test.go#L99-L168) is how I got it working.

**Comment by mostafaCamel** [comment]
Thank you!

**Comment by BenTheElder** [comment]
/remove-lifecycle stale
/transfer kubernetes
/sig api-machinery

This is the wrong place to develop client-go, it is a read-only export. We are making this clearer by closing the issue tracker and moving old issues to the repo where it is developed, there will be some other updates to make this clearer as well.

see: https://github.com/kubernetes/client-go/blob/master/CONTRIBUTING.md
